main.go

```diff
@@ -89,8 +90,17 @@ func main() {
 		log.SetOutput(f)
 	}
 
+	sigKillChan := make(chan os.Signal, 1)
+	signal.Notify(sigKillChan, os.Interrupt, os.Kill)
+
 	var wg sync.WaitGroup
```

Comment:
This WaitGroup isn't really that useful. we probably don't even need the WaitGroup internally to ProxyListener.